### PR TITLE
[KYUUBI #2782] Decouple Kyuubi Hive JDBC from Hive Serde

### LINKED ISSUE #2782: [Improvement] Decouple Kyuubi Hive JDBC from Hive Serde
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

Currently, Kyuubi Hive JDBC will convert the result from thrift type to Hive type first, then use Hive Serde to deserialize it to Java type.

### How should we improve?

Convert the data from thrift type to Java type directly, and remove `hive-serde` dependencies in Kyuubi Hive JDBC module.

Ref: https://github.com/apache/spark/pull/36373

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by jiaoqingbo**
### _Why are the changes needed?_

fix #2782 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/JdbcColumn.java [hunk 1]
```diff
@@ -68,299 +67,288 @@ public String getType() {
...
+      case ARRAY:
+      case STRUCT:
         return String.class.getName();
       default:
         throw new SQLException("Invalid column type: " + columnType);
     }
   }
 
-  static Type typeStringToHiveType(String type) throws SQLException {
+  public static int typeStringToSqlType(String type) throws SQLException {
```
Comment (yaooqinn) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/JdbcColumn.java:123]: use TType not SQL type

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiResultSetMetaData.java [hunk 2]
```diff
@@ -104,8 +104,8 @@ public boolean isAutoIncrement(int column) throws SQLException {
   public boolean isCaseSensitive(int column) throws SQLException {
     // we need to convert the Hive type to the SQL type name
     // TODO: this would be better handled in an enum
-    String type = columnTypes.get(toZeroIndex(column));
-    return "string".equalsIgnoreCase(type);
+    TTypeId type = columnTypes.get(toZeroIndex(column));
+    return true;
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiResultSetMetaData.java:108]: return true?

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/JdbcColumn.java [hunk 3]
```diff
@@ -68,43 +68,43 @@ public String getType() {
     return type;
   }
 
-  static String columnClassName(Type hiveType, JdbcColumnAttributes columnAttributes)
+  static String columnClassName(TTypeId typeStr, JdbcColumnAttributes columnAttributes)
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/JdbcColumn.java:71]: typeStr => tType

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/JdbcColumn.java [hunk 4]
```diff
@@ -113,208 +113,196 @@ static String columnClassName(Type hiveType, JdbcColumnAttributes columnAttribut
...
+        return "void";
+      default:
+        throw new SQLException("Invalid column type: " + type);
     }
-
-    throw new SQLException("Unrecognized column type: " + type);
   }
 
-  static int columnDisplaySize(Type hiveType, JdbcColumnAttributes columnAttributes)
+  static int columnDisplaySize(TTypeId typeStr, JdbcColumnAttributes columnAttributes)
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/JdbcColumn.java:220]: ditto

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiDatabaseMetaData.java [hunk 5]
```diff
@@ -741,7 +760,14 @@ public ResultSet getUDTs(
...
             "BASE_TYPE"),
-        Arrays.asList("STRING", "STRING", "STRING", "STRING", "INT", "STRING", "INT"),
+        Arrays.asList(
+            STRING_TYPE,
+            STRING_TYPE,
+            STRING_TYPE,
+            STRING_TYPE,
+            TTypeId.INT_TYPE,
+            STRING_TYPE,
+            TTypeId.INT_TYPE),
```
Comment (pan3793) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiDatabaseMetaData.java:770]: nit: style

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBaseResultSet.java [hunk 6]
```diff
@@ -46,10 +40,10 @@ public abstract class KyuubiBaseResultSet implements SQLResultSet {
   protected Object[] row;
   protected List<String> columnNames;
   protected List<String> normalizedColumnNames;
-  protected List<String> columnTypes;
+  protected List<TTypeId> columnTypes;
   protected List<JdbcColumnAttributes> columnAttributes;
 
-  private TableSchema schema;
+  private TTableSchema schema;
```
Comment (yaooqinn) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBaseResultSet.java:46]: irrelevant to this PR, maybe we can remove hive-service module too

**Review by pan3793 — APPROVED** [review]

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBaseResultSet.java (hunk 6 above)
Comment (pan3793) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBaseResultSet.java:46]: it's from hive-service-rpc, we should keep only this hive jar eventually.

Comment (jiaoqingbo) [reply kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/KyuubiBaseResultSet.java:46]: May need another pr to do this

**Comment by pan3793** [comment]
Thanks, merging to master

File: kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/JdbcColumn.java [hunk 7]
```diff
@@ -113,208 +113,196 @@ static String columnClassName(Type hiveType, JdbcColumnAttributes columnAttribut
...
-    } else if ("binary".equalsIgnoreCase(type)) {
-      return Type.BINARY_TYPE;
-    } else if ("map".equalsIgnoreCase(type)) {
-      return Type.MAP_TYPE;
-    } else if ("array".equalsIgnoreCase(type)) {
-      return Type.ARRAY_TYPE;
-    } else if ("struct".equalsIgnoreCase(type)) {
-      return Type.STRUCT_TYPE;
-    } else if ("uniontype".equalsIgnoreCase(type)) {
-      return Type.UNION_TYPE;
```
Comment (turboFei) [inline kyuubi-hive-jdbc/src/main/java/org/apache/kyuubi/jdbc/hive/JdbcColumn.java:166]: @pan3793  It is missed.

**Pull request closed without merging**